### PR TITLE
feat(argo-rollouts): Add possibility to configure healthz and metrics ports to be compatible with Istio

### PR DESCRIPTION
**Pull request by Hronom**
Add possibility to configure healthz and metrics ports to be compatible with Istio

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Review by jmeridth — CHANGES_REQUESTED** [review]
This all LGTM, just need to update the Chart.yaml version.  Then we can re-run tests and get this merged.  Thank you for your contribution.

**Comment by Hronom** [comment]
@jmeridth Updated, also I'm ok if maintainers do adjustments in this PR. So it will not be open 2 weeks because of to need bump version or whatever other reasons.

Thanks!

File: charts/argo-cd/Chart.yaml
Comment (jmeridth) [inline charts/argo-cd/Chart.yaml:5]: @Hronom this is the wrong chart file.  Needs to be `charts/argo-rollouts/Chart.yaml` 😄

**Review by jmeridth — DISMISSED** [review]

**Comment by mkilchhofer** [comment]
Hi @Hronom 
Thanks for your contribution.
Are you really sure that istio needs this on Deployment level?
I know that this can be needed on `kind: Service` but it's new to me that its also required on the Pods itself.

Aside this, can you take a look into the argo-cd chart and follow the same style for the variable names:
- Values.controller.containerPorts.metrics
- Values.controller.metrics.service.portName
- Values.controller.metrics.service.servicePort

**Comment by Hronom** [comment]
@mkilchhofer 
> Are you really sure that istio needs this on Deployment level?
I know that this can be needed on kind: Service but it's new to me that its also required on the Pods itself.

I'm just added to align names, seems to be not needed https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/

> Aside this, can you take a look into the argo-cd chart and follow the same style for the variable names:

The names that I set:
```
  - name: {{ .Values.controller.metrics.service.portName }}
    protocol: TCP
    port: {{ .Values.controller.metrics.service.port }}
    targetPort: {{ .Values.controller.metrics.service.targetPort }}
```
same as in `dashboard`:
```
  - name: {{ .Values.dashboard.service.portName }}
    protocol: TCP
    port: {{ .Values.dashboard.service.port }}
    targetPort: {{ .Values.dashboard.service.targetPort }}
```
so I assume it's standard here?

But I'm ok if maintainers change them

File: charts/argo-rollouts/Chart.yaml
Comment (yu-croco) [inline charts/argo-rollouts/Chart.yaml:26]: Please remove old change logs 🙋

File: charts/argo-rollouts/values.yaml
Comment (yu-croco) [inline charts/argo-rollouts/values.yaml:109]: Maybe better to add comment to show description on README.md? 🙋 

```diff
+# -- Controller container ports
containerPorts:
    metrics: 8090
    healthz: 8080
```

**Review by jmeridth — DISMISSED** [review]

File: charts/argo-rollouts/values.yaml
Comment (yu-croco) [inline charts/argo-rollouts/values.yaml:118]: please also add comment like https://github.com/argoproj/argo-helm/pull/1976/files#r1198905565. 🙏

**Comment by Hronom** [comment]
@yu-croco , @jmeridth added changes

**Review by mkilchhofer — COMMENTED** [review]
Thank you so much for implementing all requested changes 💯 
One last change request:
As this PR adds a feature, according to semver you need to bump the minor chart version.

As soon as its implemented, I'll approve it and merge this PR

**Comment by Hronom** [comment]
@mkilchhofer done

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution! LGTM.

**Review by mkilchhofer — APPROVED** [review]
💯

**Review by jmeridth — APPROVED** [review]

**Pull request merged**
